src/components/Todos.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from 'react';
+import { Todo } from '../types/Todo';
+import { TodoFilter } from '../enums/TodoFilters.enum';
+import { Header } from './Header';
+import { Footer } from './Footer';
+import { TodoItem } from './TodoItem';
+
+type Props = {
+  todos: Todo[];
+  tempTodo: Todo | null;
+  isLoading: boolean;
+  shouldFocus: boolean;
+  targetTodoId: number | null;
+  togglingTodoIds: number[];
+  onEmptyTitleError: () => void;
+  onAddTodo: (title: string) => void;
+  onClearCompleted: () => void;
+  onToggleOneCompleted?: (todo: Todo) => void;
+  onToggleAllCompleted: () => void;
+  onDelete: (id: number) => Promise<boolean>;
+  onRenameTodo?: (todo: Todo, title: string) => Promise<boolean>;
+};
+
+export const Todos: React.FC<Props> = ({
+  todos,
+  tempTodo,
+  isLoading,
+  shouldFocus,
+  targetTodoId,
+  togglingTodoIds,
+  onDelete,
+  onAddTodo,
+  onClearCompleted,
+  onEmptyTitleError,
+  onToggleOneCompleted,
+  onToggleAllCompleted,
+  onRenameTodo,
+}) => {
+  const [filter, setFilter] = useState<TodoFilter>(TodoFilter.All);
+
+  const hasAllTodosCompleted = todos.every(todo => todo.completed);
+  const hasCompletedTodos = todos.some(todo => todo.completed);
+  const notCompletedTodos = todos.filter(todo => !todo.completed);
+
+  const filteredTodos = todos.filter(todo => {
+    if (filter === TodoFilter.Active) {
+      return !todo.completed;
+    }
+
+    if (filter === TodoFilter.Completed) {
+      return todo.completed;
+    }
+
+    return true;
```

Comment:
use switch case here and move filter inside conditions to avoid extra iteration when no filters are selected